app/keepers/keepers.go

```diff
@@ -350,14 +325,49 @@ func NewAppKeeper(
 		appKeepers.AccountKeeper,
 		appKeepers.DistrKeeper,
 		appKeepers.BankKeeper,
-		*appKeepers.GovKeeper,
+		govkeeper.Keeper{}, // cyclic dependency between provider and governance, will be set later
 		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
 		authcodec.NewBech32Codec(sdk.GetConfig().GetBech32ValidatorAddrPrefix()),
 		authcodec.NewBech32Codec(sdk.GetConfig().GetBech32ConsensusAddrPrefix()),
 		authtypes.FeeCollectorName,
 	)
 
-	appKeepers.ProviderModule = icsprovider.NewAppModule(&appKeepers.ProviderKeeper, appKeepers.GetSubspace(providertypes.ModuleName))
+	// gov depends on provider, so needs to be set after
+	govConfig := govtypes.DefaultConfig()
+	// set the MaxMetadataLen for proposals to the same value as it was pre-sdk v0.47.x
```

Comment:
@insumity notice that this is already present before my changes

just in general, this is there to just get the same behaviour in terms of metadata length that Gaia has always had; it's just that in 47, something changed that makes it necessary to specify this now